### PR TITLE
add a default user-agent

### LINKED ISSUE #4: add a default User-Agent header
Otherwise you end up with this kind of jokes:

```txt
Request forbidden by administrative rules. Please make sure your request has a User-Agent header (http://developer.github.com/v3/#user-agent-required).
```

:)

### PR DESCRIPTION
**Pull request by zapashcanon**
@c-cube :D

fix #4 (at least it works for me and github api is happy now...)

### CODE REVIEW

File: src/core/Ezcurl_core.ml [hunk 1]
```diff
@@ -42,16 +43,16 @@ module Config = struct
   let pp out (self:t) =
     let {
       verbose; authmethod; max_redirects; follow_location;
-      username; password;
+      username; password; user_agent
     } = self in
     Format.fprintf out
       "{@[verbose=%B;@ max_redirects=%d;@ follow_location=%B;@ \
-       username=%s;@ password=%s;@ authmethod=%s@]}"
+       username=%s;@ password=%s;@ authmethod=%s; user_agent=%s@]}"
```
Comment (c-cube) [inline src/core/Ezcurl_core.ml:50]:
```suggestion
       username=%s;@ password=%s;@ authmethod=%s;@ user_agent=%s@]}"
```

File: src/core/Ezcurl_core.ml [hunk 2]
```diff
@@ -18,6 +18,7 @@ module Config = struct
     authmethod=None;
     username=None;
     password=None;
+    user_agent=Some "Ezcurl";
```
Comment (c-cube) [inline src/core/Ezcurl_core.ml:21]:
```suggestion
    user_agent=Some "curl";
```

incognito :wink:

File: src/core/Ezcurl_core.ml [hunk 3]
```diff
@@ -76,11 +77,12 @@ let delete = Curl.cleanup
 let _apply_config (self:t) (config:Config.t) : unit =
   let {
     Config.verbose; max_redirects; follow_location; authmethod;
-    username; password;
+    username; password; user_agent
   } = config in
   Curl.set_verbose self verbose;
   Curl.set_maxredirs self max_redirects;
   Curl.set_followlocation self follow_location;
+  Option.iter (fun user_agent -> Curl.set_useragent self user_agent) user_agent;
```
Comment (c-cube) [inline src/core/Ezcurl_core.ml:85]:
```suggestion
  opt_iter (fun user_agent -> Curl.set_useragent self user_agent) user_agent;
```

**Review by c-cube — APPROVED** [review]
thank you :+1:

File: src/core/Ezcurl_core.ml (hunk 2 above)
Comment (zapashcanon) [reply src/core/Ezcurl_core.ml]: @c-cube coucou-curl would have been great too ! :D

**Comment by zapashcanon** [comment]
Thansk ! @c-cube do you think you could make a new release ?

**Pull request merged**